Source/LocationManager.swift

```diff
@@ -6,283 +6,305 @@
 //  Copyright (c) 2015 Josh Adams. All rights reserved.
 //
 
-import Foundation
 import CoreLocation
 
 open class LocationManager {
-    fileprivate var gpxLocationManager: GpxLocationManager!
-    fileprivate var cLLocationManager: CLLocationManager!
+    private var gpxLocationManager: GpxLocationManager!
+    private var cLLocationManager: CLLocationManager!
+
     public enum LocationManagerType {
-        case gpx, coreLocation
+        case gpxFile(String)
+        case locations([CLLocation])
+        case cLLocationManager
```

Comment:
Is there any reason for the change from `coreLocation` to `cLLocationManager`? I kinda liked it as `coreLocation`